File: projecteuler/Euler010.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {
  public static final int MAX = 1000001;

  public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);
    int[] sieve = new int[MAX];
    Map<Integer, Long> primeSums = new HashMap<Integer, Long>();
    int t = scanner.nextInt(), k, n, i, j;
    long previousSum = 0L;

    sieve[0] = 1;
    sieve[1] = 1;
    for(i = 2; i < MAX; i++) {
      sieve[i] = 0;
    }

    for (i = 2; i < MAX; i++) {
      if (sieve[i] == 0) {
        primeSums.put(i, previousSum + i);
        previousSum += i;
        if (i <= 1000) {
          for (j = i * i; j < MAX; j += i) {
            sieve[j] = 1;
          }
        }
      } else {
        primeSums.put(i, previousSum);
      }
    }

    for (k = 0; k < t; k++) {
      System.out.println(primeSums.get(scanner.nextInt()));
    }
  }
}

File: projecteuler/euler010.rb
MAX = 1000001
prime_sums = { }
sieve = Array.new(MAX) { 0 }
sieve[0] = 1
sieve[1] = 1
i = 2
previous_sum = 0
while i < MAX do
  if sieve[i] == 0
    prime_sums[i] = previous_sum + i
    previous_sum += i
    j = i * i
    while j < MAX do
      sieve[j] = 1
      j += i
    end
  else
    prime_sums[i] = previous_sum
  end
  i += 1
end

gets.to_i.times { puts prime_sums[gets.to_i] }
